matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
You and I are working with different definitions of sparse and dense. The one used here is the same as Jason Grout defines it [here](http://ask.sagemath.org/question/9464/what-is-the-difference-between-dense-and-sparse-matrices/), namely, dense means all values are stored, and sparse means only nonzero values are stored.  Perhaps we should use the specific terminology [LIL](https://en.wikipedia.org/wiki/Sparse_matrix#List_of_lists_.28LIL.29). 

The same usage is used in the polynomials section, by the way,

>  SymPy implements dense and sparse polynomial representations. Both are used in the univariate and multivariate cases. The dense representation is the default for univariate polynomials. For multivariate polynomials, the choice of representation is based on the application. The most common case for the sparse representation is algorithms for computing Gröbner bases (Buchberger, F4, and F5). This is because different monomial orderings can be expressed easily in this representation. However, algorithms for computing multivariate GCDs or factorizations, at least those currently implemented in SymPy, are better expressed when the representation is dense. The dense multivariate representation is specifically a recursively dense representation, where polynomials in K[x0, x1, . . . , xn] are viewed as a polynomials in K[x0][x1] . . . [xn]. Note that despite this, the coefficient domain K, can be a multivariate polynomial domain as well. The dense recursive representation in Python gets inefficient when the number of variables gets high.